.github/workflows/test.yml

```diff
@@ -295,16 +299,14 @@ jobs:
             --exclude '*.h' \
             --exclude '*.output' \
             selftests/bpf/ | zstd -T0 -19 -o vmlinux-${{ matrix.arch }}-${{ matrix.toolchain_full }}.tar.zst
-      - if: ${{ github.event_name != 'push' }}
-        name: Remove KBUILD_OUTPUT contents
+      - name: Remove KBUILD_OUTPUT contents
```

Comment:
Can you please clarify why this change is necessary? It strikes me as wrong. Now we are no longer creating caches at all, right? I.e., it breaks incremental builds.
Also, comment in line 305 is outdated now.

Reply:
This might be unnecessary indeed. I think I read it as '==' for some reason.
Thanks for all the comments, I'll reply in each after making changes and re-testing.